src/cursor/cur_std.c

```diff
@@ -651,6 +651,8 @@ __wt_cursor_cache(WT_CURSOR *cursor, WT_DATA_HANDLE *dhandle)
 
     WT_TRET(cursor->reset(cursor));
 
+    __wt_cursor_bound_reset(cursor);
```

Comment:
Add a comment above all callers to this function just so its clear why its here.